go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -91,16 +91,17 @@ def update_files # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity
             # Replace full paths with path hashes in the go.mod
             substitute_all(substitutions)
 
-            # Set the stubbed replace directives
-            update_go_mod(dependencies)
-
-            # Then run `go get` to pick up other changes to the file caused by
-            # the upgrade
-            run_go_get
+            # TODO as far as I can tell, this array will only ever have one value, but we should assert that somehow
+            # and blow up if it's ever not true... I'm not a rubyist, so not sure how best to handle?
+            dep = dependencies.first
+            # `go get` will generate the required go.mod/go.sum updates for the new dep version
+            run_go_get(dep)
 
             # If we stubbed modules, don't run `go mod {tidy,vendor}` as
             # dependencies are incomplete
             if substitutions.empty?
+              # go mod tidy should run before go mod vendor to ensure any
+              # dependencies removed by go mod tidy are also removed from vendors.
```

Comment:
I ran across this comment on an old PR while researching my change... the comment seems very useful/easy to miss, so thought I should add it here. If needed I can break it out as a separate PR.

Reply:
I broke this out as https://github.com/dependabot/dependabot-core/pull/3732 for simplicity.